grizzly-ls/tests/unit/test___main__.py

```diff
@@ -146,8 +179,81 @@ def test_setup_logging(mocker: MockerFixture, capsys: CaptureFixture[str]) -> No
     assert capture.out == ''
 
 
+def test_setup_debugging(mocker: MockerFixture, capsys: CaptureFixture[str]) -> None:
+    mocker.patch('grizzly_ls.__main__.logging.info')
+    logging_error_mock = mocker.patch('grizzly_ls.__main__.logging.error')
+
+    # enable debugging with missing debugpy module
+    prev_path = sys.path
+    if 'debugpy' in sys.modules:
+        del sys.modules['debugpy']
+    sys.path = []
+
+    arguments = Namespace(
+        socket=False,
+        verbose=False,
+        no_verbose=None,
+        debug=True,
+        debug_port=5678,
+        debug_wait=False,
+    )
+    setup_debugging(arguments)
+    logging_error_mock.assert_called_once_with('Debugging requires the debugpy package to be installed')
+    sys.path = prev_path
```

Comment:
maybe the restore part should be in a `finally` block so it's always done, even if the test for some reason fails.